tests/unit/io/Cereal_unittest.cc

```diff
@@ -0,0 +1,31 @@
+#include <shogun/lib/common.h>
+#include <shogun/base/Parameter.h>
+#include <gtest/gtest.h>
+
+#include <shogun/lib/SGVector.h>
+#include <cereal/archives/json.hpp>
+#include <fstream>
+
+using namespace shogun;
+
+TEST(Cereal, Json_SGVector_load_equals_saved)
+{
+	const index_t size = 5;
+	SGVector<float64_t> a(size), b;
+	a.range_fill(1.0);
+
+	{
+		std::ofstream os("cereatltest_sgvector.cereal");
+		cereal::JSONOutputArchive archive(os);
+		archive(a);
+	}
+
+	{
+		std::ifstream is("cereatltest_sgvector.cereal");
+		cereal::JSONInputArchive archive(is);
+		archive(b);
+	}
+
+	for (index_t i = 0; i < size; i++)
+		EXPECT_NEAR(a[i], b[i], 1E-15);
+}
```

Comment:
please make sure that you delete the `cereatltest_sgvector.cereal` file before exiting this function:
 * anticipate IO exceptions...
 * please use temporary files (see the other serialization unittests) to avoid any possible inconsistent state...